frontend/src/components/BatchTransfer.tsx

```diff
@@ -20,108 +20,110 @@ import { userAccountState } from '../store';
 import { logout, sendFT, getTxChannel, getBalances } from '../services/flow';
 import ConfirmTable from '../components/ConfirmTable';
 import SendButton from './SendButton';
-
-type Currency = {
-  symbol: string;
-  contractName: string;
-  address: string;
-  vaultStoragePath: string;
-  vaultPublicPath: string;
-};
-
-const FLOWCurrency: Currency = {
-  symbol: 'FLOW',
-  contractName: 'FlowToken',
-  address:
-    process.env.NEXT_PUBLIC_NETWORK == 'mainnet'
-      ? '0x1654653399040a61'
-      : '0x7e60df042a9c0868',
-  vaultStoragePath: '/storage/flowTokenVault',
-  vaultPublicPath: '/public/flowTokenReceiver',
-};
-
-const CustomCurrency: Currency = {
-  symbol: '',
-  contractName: '',
-  address: '',
-  vaultStoragePath: '',
-  vaultPublicPath: '',
-};
-
-const FUSDCurrency: Currency = {
-  symbol: 'FUSD',
-  contractName: 'FUSD',
-  address:
-    process.env.NEXT_PUBLIC_NETWORK == 'mainnet'
-      ? '0x3c5959b568896393'
-      : '0xe223d8a629e49c68',
-  vaultStoragePath: '/storage/fusdVault',
-  vaultPublicPath: '/public/fusdReceiver',
-};
+import { ValidationError } from 'types/error';
+import { CustomCurrency, FLOWCurrency, FUSDCurrency } from 'types/currency';
+import { Output } from 'types/transaction';
 
 const explorerUrl =
   process.env.NEXT_PUBLIC_NETWORK == 'mainnet'
     ? 'https://flowscan.org/transaction/'
     : 'https://testnet.flowscan.org/transaction/';
 
+const isValidAddress = (address: string): boolean => {
+  if(!address) {
+    return false;
+  }
+  return !!address.match(/^0x[0-9a-f]{16}$/)
+}
+
 const BatchTransfer = () => {
   const {
     handleSubmit,
     formState: { errors, isSubmitting },
   } = useForm();
   const [userAccount, setUserAccount] = useRecoilState(userAccountState);
   const [currency, setCurrency] = useState(FLOWCurrency);
-  const [recipientTemplate, setRecipientTemplate] = useState(''); // text area
+  const [txTemplate, setTxTemplate] = useState(''); // text area
```

Comment:
This variable means information about the recipients and the amounts. The original name is not good, but `txTemplate` is also a bit strange.

Reply:
txTemplate -> outputsTemplate
https://github.com/flow-japan/batch-token-transfer-tool/pull/3/commits/5681064770206e23da7c75ac8a79f773f583b019